### PR TITLE
UIP-3088 Fix detection of packages whose names contain numbers

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
## Ultimate Problem
The regex that detected import/export statements didn't account for packages with numbers in their names.

This can cause false positives and false negatives when using dependnency_validator (I hit one of each when using it).

## Solution
- Account for numbers in package names
- Add tests for package name regex, including regression tests for package names with numbers
- Boyscouting: update regex for two additional edge cases:
    - Leading spaces
    - Triple-quoted exports (apparently you can do that)
    - imports/exports not separated by newlines
        - This is definitely an edge case, so let me know if my changes to the regex to support this too severe
    - Note: it looks like adjacent string concatenation is also allowed; supporting that is probably best left for another time (if at all), and best for a tool like `analyzer` as opposed to regex.

## Testing
Verify tests pass

## Areas of Regression
Detection of packages.

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Module Findings
No security relevant content detected. Please review for security relevance and request security review as needed.

File: test/import_export_regex_test.dart [hunk 1]
```diff
@@ -0,0 +1,78 @@
+// Copyright 2017 Workiva Inc.
```
Comment (toddbeckman-wf) [inline test/import_export_regex_test.dart]: 2018

Comment (greglittlefield-wf) [reply test/import_export_regex_test.dart]: This is present in all the files; I'll make a separate PR following this to update them if that's cool.

**Comment by toddbeckman-wf** [comment]
+10

- [x] Tests exercise regex
- [x] CI passes

**Comment by maxwellpeterson-wf** [comment]
QA +1
- [x] Dev +1's
- [x] Dev/QA +10 with detail of what was tested
  - https://github.com/Workiva/dependency_validator/pull/17#issuecomment-369408078
- [x] Tests created/updated 
- [x] All tests pass

@Workiva/release-management-pp
